nzshm_common/location/code_location.py

```diff
@@ -13,14 +13,22 @@ class CodedLocation:
     lon: float = field(hash=True)
     resolution: float = field(hash=True)
 
-    def __init__(self, lat, lon, resolution):
+    def __init__(self, lat: float, lon: float, resolution: float) -> None:
+        """
+        Create a CodedLocation instantce.
+
+        Arguments:
+            lat: latitude
+            lon: longitude
+            resolution: the resolution used to resolve the location
+        """
         assert 0 < resolution < 180
```

Comment:
You can add an explanation to assertions, e.g.

```py
assert 0 < resolution < 180, "Resolution must be between 0 and 180 degrees."
```

Which will yield an error that looks like:

```
AssertionError: Resolution must be between 0 and 180 degrees.
```

Reply:
OK - nice, let's do more of that!